app.js

```diff
@@ -88,26 +65,77 @@ const bodyParserErrorHandler = (err, req, res, next) => {
 };
 
 /**
- * @summary Initialize API with OpenAPI specification
+ * @summary Function that handles transforming openapi errors
  */
-initialize({
-  app: appRouter,
-  apiDoc: openapi,
-  paths: `${appRoot}/api/v1/paths`,
-  consumesMiddleware: {
-    'application/json': compose([bodyParser.json(), bodyParserErrorHandler]),
-  },
-  errorMiddleware: errors([openAPIErrorMiddleware, genericErrorMiddleware]),
-  errorTransformer: (openapiError, ajvError) => Object.assign({}, openapiError, ajvError),
-});
+const errorTransformer = (openapiError, ajvError) => {
+  /**
+   * express-openapi will add a leading '[' and closing ']' to the 'path' field if the parameter
+   * name contains '[' or ']'. This regex is used to remove them to keep the path name consistent.
+   * @type {RegExp}
+   */
+  const pathQueryRegex = /\['(.*)']/g;
 
-/**
- * @summary Return a 404 error if resource not found
- */
-appRouter.use((req, res) => errorBuilder(res, 404, 'Resource not found.'));
+  const error = Object.assign({}, openapiError, ajvError);
+
+  const regexResult = pathQueryRegex.exec(error.path);
+  error.path = regexResult ? regexResult[1] : error.path;
+  return error;
+};
 
 /**
- * @summary Start servers and listen on ports
+ * @summary Handles general startup for the app
  */
-httpsServer.listen(serverConfig.port);
-adminHttpsServer.listen(serverConfig.adminPort);
+const startup = async () => {
+  await SwaggerParser.validate('openapi.yaml').then((openapi) => {
+    app.locals.openapi = openapi;
```

Comment:
Can you add some comment to address why we need `locals` and when can we use it? Since it looks like we'll only store the variables to `app.locals` during the `startup`, I think you can initialize a local variable `app.locals` at the beginning of this function and add some comments there.

Reply:
From our discussion, assigning `app.locals` to a local variable might cause some issues since there will be some initial fields in `app.locals` that will get overwritten. Instead, I think I could add some documentation for the block of code that sets `app.locals.openapi` for clarity.